pypeit/coadd2d.py

```diff
@@ -1589,7 +1598,8 @@ def compute_weights(self, weights):
             for id in slitord_ids:
                 _, iweights = self.optimal_weights(id, self.objid_bri)
                 use_weights.append(iweights)
-            self.use_weights = np.array(use_weights)
+            # Commenting this out since self.use_weights needs to be a list or a string
+            #self.use_weights = np.array(use_weights)
```

Comment:
Hey @jhennawi , maybe I was not clear here. The core routine takes `self.use_weights` and you are commenting it out here (right where it is defined in the case when `weights = 'auto'`). It's fine it is a list, but you need to assign the list to `self.use_weights`. You are passing the tests because the default is None (see `__init__`) and the core routine takes it, but the code is not doing what you want it to do.

Reply:
Same comment as above. Please fix as you see fit, and check with tests.